### DISCUSSION TITLE
Issue in Node

### DISCUSSION THREAD

**Discussion by yuvrajsagar17** [community/community#64544] [Programming Help] [answered]
### Body

Hello fellow developers,
I hope you're having a great day! I'm currently working on a project using Node.js, and I'm running into a bit of trouble with event handling. I'm trying to create a custom event emitter and listener, but it seems like I'm missing something in my code. Here's a simplified version of what I have:

```
const EventEmitter = require('events');
class MyEmitter extends EventEmitter {}
const myEmitter = new MyEmitter();
myEmitter.on('customEvent', () => {
    console.log('Custom event triggered');
});

myEmitter.emit('customEvent');
```

However, I'm not seeing the 'Custom event triggered' message in my console as expected. Could someone please point me in the right direction? Any insights would be really helpful!

Thank you for your time and assistance!

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by M-YasirGhaffar** [comment, marked as answer]
One common mistake when working with the EventEmitter in Node.js is that the event listener might not get triggered if the event is emitted before the listener is attached. In your code, you're emitting the 'customEvent' immediately after attaching the listener, so it's possible that the event is being emitted before the listener is fully set up.

To ensure that the listener is in place before the event is emitted, you can try moving the emit statement inside a setTimeout function with a slight delay, like this:

javascript
Copy code
const EventEmitter = require('events');
class MyEmitter extends EventEmitter {}
const myEmitter = new MyEmitter();

myEmitter.on('customEvent', () => {
    console.log('Custom event triggered');
});

setTimeout(() => {
    myEmitter.emit('customEvent');
}, 100); // Adjust the delay as needed
This should give the listener enough time to attach before the event is emitted. If you still don't see the message in your console, please double-check your environment and dependencies to make sure everything is set up correctly.

Remember that this is just one potential solution, and there might be other factors at play. Feel free to experiment with your code and environment, and don't hesitate to ask if you have any more questions or run into any issues. Happy coding!
